samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c

```diff
@@ -0,0 +1,312 @@
+
+struct dect_phy_mac_nbr_bg_scan_metrics_data {
+	uint32_t scan_started_ok_count;
+	uint32_t scan_start_fail_count;
+	uint32_t scan_info_updated_count;
+	uint32_t scan_info_time_shift_updated_count;
+	int64_t scan_info_time_shift_last_value;
+};
+
+struct dect_phy_mac_nbr_bg_scan_data {
+	bool running;
+	struct dect_phy_mac_nbr_bg_scan_params params;
+	struct dect_phy_mac_nbr_bg_scan_metrics_data metrics;
+
+	uint64_t last_updated_rcv_time;
+};
+
+static struct dect_phy_mac_nbr_bg_scan_data nbr_bg_scan_data[DECT_PHY_MAC_MAX_NEIGBORS];
+
+static struct dect_phy_mac_nbr_bg_scan_data *dect_mac_nbr_bg_scan_free_list_item_get(void)
+{
+	for (int i = 0; i < DECT_PHY_MAC_MAX_NEIGBORS; i++) {
+		if (!nbr_bg_scan_data[i].running) {
+			return &nbr_bg_scan_data[i];
+		}
+	}
+	return NULL;
+}
+
+static struct dect_phy_mac_nbr_bg_scan_data *
+dect_mac_nbr_bg_scan_list_item_by_phy_op_handle_get(uint32_t phy_op_handle)
+{
+	for (int i = 0; i < DECT_PHY_MAC_MAX_NEIGBORS; i++) {
+		if (nbr_bg_scan_data[i].running &&
+		    nbr_bg_scan_data[i].params.phy_op_handle == phy_op_handle) {
+			return &nbr_bg_scan_data[i];
+		}
+	}
+	return NULL;
+}
+
+static struct dect_phy_mac_nbr_bg_scan_data *
+dect_mac_nbr_bg_scan_list_item_by_nbr_long_rd_id_get(uint32_t long_rd_id)
+{
+	for (int i = 0; i < DECT_PHY_MAC_MAX_NEIGBORS; i++) {
+		if (nbr_bg_scan_data[i].running &&
+		    nbr_bg_scan_data[i].params.target_long_rd_id == long_rd_id) {
+			return &nbr_bg_scan_data[i];
+		}
+	}
+	return NULL;
+}
+
+static void dect_phy_mac_nbr_bg_scan_scheduler_op_to_mdm_cb(
+	struct dect_phy_common_op_completed_params *params, uint64_t frame_time)
+{
+	if (params->status == NRF_MODEM_DECT_PHY_SUCCESS) {
+		dect_phy_ctrl_msgq_non_data_op_add(DECT_PHY_CTRL_OP_DEBUG_OFF);
+	}
+}
+
+static void dect_phy_mac_nbr_bg_scan_scheduler_op_completed_cb(
+	struct dect_phy_common_op_completed_params *params, uint64_t tx_frame_time)
+{
+	/* If we haven't received from a beacon, stop a bg scan and call a callback */
+	struct dect_phy_mac_nbr_bg_scan_data *bg_scan_data =
+		dect_mac_nbr_bg_scan_list_item_by_phy_op_handle_get(params->handle);
+	if (!bg_scan_data) {
+		desh_error("(%s): bg_scan_data not found (handle %d)", (__func__), params->handle);
+		return;
+	}
+	uint64_t time_now = dect_app_modem_time_now();
+	int64_t time_from_last_received_ms =
+		MODEM_TICKS_TO_MS(time_now - bg_scan_data->last_updated_rcv_time);
+
+	if (bg_scan_data->last_updated_rcv_time != 0 &&
+	    time_from_last_received_ms > DECT_PHY_MAC_NBR_BG_SCAN_MAX_UNREACHABLE_TIME_MS) {
+		bg_scan_data->running = false;
+		dect_phy_api_scheduler_list_item_remove_by_phy_op_handle(params->handle);
+		if (bg_scan_data->params.cb_op_completed) {
+			bg_scan_data->params.cb_op_completed(&(
+				struct dect_phy_mac_nbr_bg_scan_op_completed_info){
+				.cause = DECT_PHY_MAC_NBR_BG_SCAN_OP_COMPLETED_CAUSE_UNREACHABLE,
+				.phy_op_handle = params->handle,
+				.target_long_rd_id = bg_scan_data->params.target_long_rd_id,
```

Comment:
I don't know if this way of passing a structure was used elsewhere but I guess we could introduce it as a parameter before the call and pass it in a bit more normal manner?

Reply:
I was surprised that this worked with C-language like this. OK, I'll change these. from this file.